adagrams/game.py

```diff
@@ -1,11 +1,130 @@
+import random as r
+
+#Define the frequencies of each letter
+LETTER_POOL = {
+'A': 9,
+'B': 2,
+'C': 2,
+'D': 4,
+'E': 12,
+'F': 2,
+'G': 3,
+'H': 2,
+'I': 9,
+'J': 1,
+'K': 1,
+'L': 4,
+'M': 2,
+'N': 6,
+'O': 8,
+'P': 2,
+'Q': 1,
+'R': 6,
+'S': 4,
+'T': 6,
+'U': 4,
+'V': 2,
+'W': 2,
+'X': 1,
+'Y': 2,
+'Z': 1
+}
+
+#Define score for each letter
+SCORE_DICT = {
+    "A": 1,
+    "B": 3,
+    "C": 3,
+    "D": 2,
+    "E": 1,
+    "F": 4,
+    "G": 2,
+    "H": 4,
+    "I": 1,
+    "J": 8,
+    "K": 5,
+    "L": 1,
+    "M": 3,
+    "N": 1,
+    "O": 1,
+    "P": 3,
+    "Q": 10,
+    "R": 1,
+    "S": 1,
+    "T": 1,
+    "U": 1,
+    "V": 4,
+    "W": 4,
+    "X": 8,
+    "Y": 4,
+    "Z": 10
+}
+
+def make_letter_list(dict):
+    #Turn dict into list where each letter appears a number of times
+    #equal to its frequency
+    letter_list = []
+    for letter, count in dict.items():
+        letter_list.extend([letter for _ in range(count)])
+
+    return letter_list
+
+
 def draw_letters():
-    pass
+    letter_list = make_letter_list(LETTER_POOL)
+
+    #Create an empty hand, then add 10 random letters from letter list.
+    #Letters are removed from list as they are added to hand.
+    hand = []
+
+    for _ in range(10):
+        chosen_letter = r.choice(letter_list)
+        hand.append(chosen_letter)
+        letter_list.remove(chosen_letter)
```

Comment:
Ruminating on this...what I am thinking right now is using randint instead of choice. You could designate a chunk of the end of the list to be "already-chosen" letters (using a variable to keep track of the position of the end of the "unchosen" letters in the list), and on each iteration of the loop you can swap the item at the chosen index (after adding it to the hand) to the end of the "unchosen" letters and update the variable you're using to indicate the high end of the values for randint. Not 100% sure this is good (or whether I'm explaining it well) but I think that would be more efficient.